### ISSUE TITLE
Add funding streams to regtest

### ISSUE THREAD

**Issue by josef-v** [zcash/zcash#4681]
There are currently no funding streams present in regtest. This would be expected as founders reward is present before canopy activation and funding streams are expected to be the mechanism replacing founders reward.

The funding streams should be active from the block the canopy upgrade is active (set by nuparams) for a set amount of blocks (post blossom halving interval to be consistent).

Rationale:
- it allows testing of code that builds coinbase transactions from scratch in continuous integration
- it's a major feature and it makes sense to have it in regtest

**Comment by str4d** [comment]
Canopy has now activated, so this is lower priority, but it might still be useful. More generally, it would be good to use this as a test case for how this kind of request might have been added if #3722 were implemented in the codebase (which it will be ahead of the next NU, once #4480 is merged).

**Comment by josef-v** [comment]
Just some insights on this issue:
We have made some patches adding funding streams to regtest ourselves. This was done to test that we can handle the transition from non-canopy to canopy in all of our infrastructure (stratum servers, rewards calculations). Test was done at arbitrary height which was low enough to make the test quick. Other stuff like testing of rotation of funding streams addresses was not part of this test (and couldn't be as explained in another part of this comment) and was tested using mainnet's `getblocksubsidy`.
I have encountered following problems:
- Height of funding stream activation is fixed (without adding some more complex logic) and is independent on setting of canopy activation by nuparams. This was easy to solve for our one shot test as nothing else than deciding at which block to start funding streams and canopy was needed.
- Logic and calculations surrounding funding streams are more complex in implementation than in documentation. It works the same for mainnet and testnet, however trying to start canopy at some arbitrary height to allow quick testing (let's say block 2) is not that easy. As we only wanted to test transition from pre-canopy to post-canopy, I have simply removed the asserts testing that funding streams addresses won't run out (and other similar asserts). Sure, this would not work for a longer running testing chain, but for our case, this was sufficient.

This means that adding funding streams to regtest and also allowing setting of canopy upgrade height by nuparams is not that easy and would either need to enforce some limitations on activation height or addition of some more complex logic on handling funding streams at arbitrary activation height.
